app/models/case_log.rb

```diff
@@ -35,7 +35,9 @@ class CaseLog < ApplicationRecord
   default_scope -> { kept }
 
   validates_with CaseLogValidator
-  before_save :update_status!
+  before_validation :update_status!
```

Comment:
Should this not still happen `before_save`? If validation fails we wouldn't have expected the status to change?

Reply:
Good point, this needs to be changed.